File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        sort_nums = list(sorted(set(nums)))
        res = 0
        i = 1
        consecutive = 1

        while i < len(sort_nums):
            if sort_nums[i] == sort_nums[i-1] + 1:
                consecutive += 1
            else:
                res = max(res, consecutive)
                consecutive = 1
            i += 1
        res = max(res, consecutive)
        return res

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() < 1) return 0;

        sort(nums.begin(), nums.end());
        
        int res = 1, consecutive = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i-1] + 1) {
                consecutive++;
            } else if (nums[i] == nums[i-1]) {
                continue;
            } else {
                res = max(res, consecutive);
                consecutive = 1;
            }
        }
        res = max(res, consecutive);
        return res;
    }
};